### PR TITLE
Shopping cart was added, code slightly refactored

### PR DESCRIPTION
**Pull request by MaxSkory**

### CODE REVIEW

File: src/main/java/mskory/bookstore/model/ShoppingCart.java [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+import lombok.Data;
+import lombok.EqualsAndHashCode;
+import lombok.NoArgsConstructor;
+import lombok.ToString;
+
+@Data
+@Entity
+@Builder
+@AllArgsConstructor
+@NoArgsConstructor
```
Comment (Vasul-Pankiv) [inline src/main/java/mskory/bookstore/model/ShoppingCart.java]: if you use @Builder it is useless to add @AllArgsConstructor and @NoArgsConstructor until you create your own constructor

File: src/main/java/mskory/bookstore/model/ShoppingCart.java [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+    @OneToMany(fetch = FetchType.LAZY, mappedBy = "shoppingCart",
+            cascade = CascadeType.ALL, orphanRemoval = true)
+    @ToString.Exclude
+    @EqualsAndHashCode.Exclude
+    private Set<CartItem> cartItems;
+
+    public void addCartItem(CartItem cartItem) {
+        cartItems.add(cartItem);
+        cartItem.setShoppingCart(this);
+    }
```
Comment (Vasul-Pankiv) [inline src/main/java/mskory/bookstore/model/ShoppingCart.java]: move it to the service class

**Review by Vasul-Pankiv — CHANGES_REQUESTED** [review]
Great work!!!

File: src/main/java/mskory/bookstore/model/ShoppingCart.java (hunk 1 above)
Comment (MaxSkory) [reply src/main/java/mskory/bookstore/model/ShoppingCart.java]: @Builder annotation needs both of @AllArgsConstructor and @NoArgsConstructor as i know. Am I not right?

**Review by olekskov — APPROVED** [review]

**Pull request merged**